### PR TITLE
Blog: How to Monitor and Optimize Node-RED Flows Using Open Telemetry

### PR DESCRIPTION
**Pull request by sumitshinde-84**
…RED Flows

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] I have considered the performance impact of these changes
 - [ ] Suitable unit/system level tests have been added and they pass
 - [ ] Documentation has been updated

### CODE REVIEW

File: src/blog/2024/08/opentelemetry-with-node-red.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+// Convert Kelvin to Celsius
+let celsius = kelvinToCelsius(kelvin);
+
+// Set the result in the message payload
+msg.payload = celsius;
+
+// Return the modified message object
+return msg;
+```
```
Comment (ZJvandeWeg) [inline src/blog/2024/08/opentelemetry-with-node-red.md:98]: What's the low code equivalent? Can't the chnage node do this?

File: src/blog/2024/08/opentelemetry-with-node-red.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   - posts
+   - flowfuse
+   - node-red
+   - distributed tracing in node-red
+   - open telemetry with node-red
+   - optimizing the performance of node-red flows
+   - how to monitor performance of node-red flows
+---
+
+Have you ever found yourself frustrated by unexpected delays in your Node-RED flows, wondering where the bottlenecks are hiding? In the fast-paced world of IoT, even small latency issues can have a big impact on your system's performance. That's where Open Telemetry comes in. With its powerful distributed tracing capabilities, you can finally take control and get a clear view of how your flows are performing in real time.
```
Comment (ZJvandeWeg) [inline src/blog/2024/08/opentelemetry-with-node-red.md:18]: Less ChatGPT please -- "In the fast-paced world of IoT, even small latency  ..."

File: src/blog/2024/08/opentelemetry-with-node-red.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+<!--more-->
+
+In this guide, we’ll show you how to integrate Open Telemetry with Node-RED to monitor latency across your flows. By implementing distributed tracing, you’ll gain the ability to see exactly where delays occur, helping you optimize performance and ensure your IoT applications run efficiently.
```
Comment (ZJvandeWeg) [inline src/blog/2024/08/opentelemetry-with-node-red.md:22]:
```suggestion
Integrating Open Telemetry with Node-RED allows you to monitor latency across your flows. By implementing distributed tracing, you’ll gain the ability to see exactly where delays occur, helping you optimize performance and ensure your IoT applications run efficiently.
```

File: src/blog/2024/08/opentelemetry-with-node-red.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+{% endrenderFlow %}
+
+Deploy the above flow, and you might notice that the data printed on the debug panel appears with a delay. For demonstration purposes, we have intentionally added a delay in the function node that converts temperature data to Celsius. While this delay might be easy to spot in a simple flow, identifying such issues in larger flows with hundreds of nodes can be challenging and time-consuming. OpenTelemetry simplifies this task by providing detailed traces that help you pinpoint where delays or issues are occurring.
```
Comment (ZJvandeWeg) [inline src/blog/2024/08/opentelemetry-with-node-red.md:42]: @sumitshinde-84 There's literally a delay node -- Why use High code here?

File: src/blog/2024/08/opentelemetry-with-node-red.md (hunk 1 above)
Comment (sumitshinde-84) [reply src/blog/2024/08/opentelemetry-with-node-red.md]: Done! I have replaced the function node with a change node and added a delay node for the delay.

**Review by ZJvandeWeg — APPROVED** [review]

**Comment by sumitshinde-84** [comment]
Title is too long. Updating it to: Monitoring and Optimizing Node-RED Flows with OpenTelemetry.

**Comment by sumitshinde-84** [comment]
wait !

**Pull request merged**